File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/Div.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitable;
import org.uva.sea.ql.ast.visit.Visitor;

public class Div extends BinaryExpr implements Visitable {
	
	public Div(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
	}
	
	@Override
	public Integer eval() {
		return (Integer) lhs.eval() / (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor) {
		visitor.visit(this, null);
	}

	@Override
	public String toString() {
		return super.toString() + " (/)";
	}
	
	@Override
	public Type getType(Context context) {
		return Type.INT;
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/div.go
package binaryoperatorexpr

import "ql/ast/expr"

type Div struct {
	BinaryOperator
}

func NewDiv(lhs expr.Expr, rhs expr.Expr) Div {
	return Div{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (d Div) Eval(s interface{}) interface{} {
	return d.Lhs.Eval(s).(int) / d.Rhs.Eval(s).(int)
}
